### PR TITLE
[CI] Add static checks in github action and sanitizers for unit tests

### PR DESCRIPTION
**Pull request by czm23333**
### What problem does this PR solve?

Issue Number: #2570

Problem Summary: Use github action to do some static checks and code security tests on the code in the curve warehouse

### What is changed and how it works?

What's Changed: Github action workflow, .bazelrc, ut.sh and util/ut_in_image.sh

How it Works: By adding some jobs in github action workflow, cppcheck and cpplint is now automatically performed on push and pull request, and results are published to action files.

The .bazelrc is changed to add building configs for enabling ASan, MSan and UBSan. Then running ut.sh with "--sanitizer [mode]" will use corresponding config to build curve with sanitizers and run tests (specifically, mode=1 for ASan and mode=2 for MSan and UBSan).

Side effects(Breaking backward compatibility? Performance regression?): Running tests with sanitizers may be ~2x-4x slower.

### Check List

- [Y] Relevant documentation/comments is changed or added
- [Y] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by NaturalSelect** [comment]
cicheck

**Comment by czm23333** [comment]
cicheck

**Comment by czm23333** [comment]
cicheck

**Comment by czm23333** [comment]
cicheck

**Review by YunhuiChen — APPROVED** [review]

File: .bazelrc [hunk 1]
```diff
@@ -7,4 +7,20 @@ build:gcc7-later --cxxopt -faligned-new
...
+build:asan --linkopt -fsanitize=address
+
+build:msan --strip=never
+build:msan --copt -fsanitize=memory
+build:msan --copt -fsanitize=undefined
+build:msan --copt -O1
+build:msan --copt -g
+build:msan --copt -fno-omit-frame-pointer
+build:msan --linkopt -fsanitize=address
+build:msan --linkopt -fsanitize=undefined
```
Comment (wu-hanqing) [inline .bazelrc:24]:
```suggestion
build:sanitize-common --strip=never
build:sanitize-common --copt -O1
build:sanitize-common --copt -g
build:sanitize-common --copt -fno-omit-frame-pointer

build:asan --config=sanitize-common
build:asan --copt -fsanitize=address
build:asan --copt -DADDRESS_SANITIZER
build:asan --linkopt -fsanitize=address

build:asan --config=sanitize-common
build:msan --copt -fsanitize=memory
build:msan --copt -fsanitize=undefined
build:msan --linkopt -fsanitize=address
build:msan --linkopt -fsanitize=undefined
```

File: .github/workflows/cppcheck.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: cppcheck-action
```
Comment (wu-hanqing) [inline .github/workflows/cppcheck.yml:1]: This test fails.

Comment (czm23333) [reply .github/workflows/cppcheck.yml:1]: > This test fails.

It actually succeeds.

https://github.com/czm23333/curve/actions/runs/6058965735/job/16441596764

I have no idea why it fails here.

**Comment by czm23333** [comment]
cicheck

**Comment by czm23333** [comment]
cicheck

**Pull request closed without merging**